src/preset.ts

```diff
@@ -572,43 +651,9 @@ async function expandConfigurePresetHelper(folder: string,
                                                                     shell: true }).result;
         if (!clLoc.stdout) {
           // Not on PATH, need to set env
-          let arch = 'x86';
-          let toolsetArch = 'host=x86';
-          let toolsetVsVersion: string | undefined;
-          if (util.isString(preset.architecture)) {
-            arch = preset.architecture;
-          } else if (preset.architecture && preset.architecture.value) {
-            arch = preset.architecture.value;
-          } else {
-            log.warning(localize('no.cl.arch', 'Configure preset {0}: No architecture specified for cl.exe, using x86 by default', preset.name));
-          }
-          const toolsetArchRegex = /(host=\w+),?/i;
-          const toolsetVsVersionRegex = /version=(\w+),?/i;
-          const noToolsetArchWarning = localize('no.cl.toolset.arch', 'Configure preset {0}: No toolset architecture specified for cl.exe, using x86 by default', preset.name);
-          const noToolsetVsVersionWarning = localize('no.cl.toolset.version', 'Configure preset {0}: No toolset version specified for cl.exe, using latest by default', preset.name);
-          const matchToolsetArchAndVersion = (toolset: string) => {
-            const toolsetArchMatches = toolset.match(toolsetArchRegex);
-            if (!toolsetArchMatches) {
-              log.warning(noToolsetArchWarning);
-            } else {
-              toolsetArch = toolsetArchMatches[1];
-            }
-            const toolsetVsVersionMatches = toolset.match(toolsetVsVersionRegex);
-            if (!toolsetVsVersionMatches) {
-              log.warning(noToolsetVsVersionWarning);
-            } else {
-              toolsetVsVersion = toolsetVsVersionMatches[1];
-            }
-          };
-          if (!preset.toolset) {
-            log.warning(noToolsetArchWarning);
-          } else if (util.isString(preset.toolset)) {
-            matchToolsetArchAndVersion(preset.toolset);
-          } else if (!preset.toolset.value) {
-            log.warning(noToolsetArchWarning);
-          } else {
-            matchToolsetArchAndVersion(preset.toolset.value);
-          }
+          const arch = getArchitecture(preset);
+          const toolset = getToolset(preset);
```

Comment:
I can not hit this code. the cl location is found and then I won't see any error related to toolset.
Could you please share your `configurePresets`?

Reply:
I believe these were the important parts to get into this situation:
```
{
  "generator": "Ninja",
  "architecture": { ... },
  "toolset": { ... },
  "cacheVariables": {
    "CMAKE_C_COMPILER": "cl.exe",
    "CMAKE_CXX_COMPILER": "cl.exe"
  }
}
```

Reply:
I still need to run the code like this:
```
        const arch = getArchitecture(preset);
        const toolset = getToolset(preset);
        if (!clLoc.stdout) {
```

to be able to see the warning messages. 

![image](https://user-images.githubusercontent.com/16143955/124661233-12e09e00-de5c-11eb-82fe-8702cc0ff1f5.png)